jupyterlab_comments/jupyterlab_comments/handlers_test.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import os
+import tempfile
+import handlers
+import subprocess
+import unittest
+import json
+import tornado.ioloop
+import tornado.web
+import tornado.testing
+
+remote_dir = tempfile.TemporaryDirectory()
```

Comment:
We aren't cleaning up the temporary directories anywhere, which means they will stick around until the machine running the test reboots.

We should take advantage of the fact that these return context managers so that we use them in a `with` statement. That way, the temporary director(y|ies) get automatically deleted after the `with` statement completes.

If you want to use multiple directories you can do this like:

```py
    with tempfile.TemporaryDirectory() as remote_dir, \
           tempfile.TemporaryDirectory() as not_get_dir, \
           tempfile.TemporaryDirectory() as dir1, \
           tempfile.TemporaryDirectory() as dir2:
       ...
```

Alternatively, you can use a single temporary directory as a parent and create all the other directories as regular directories under that:

```py
    with tempfile.TemporaryDirectory() as working_dir:
       remote_dir = os.path.join(working_dir, "remote")
       os.makedirs(remote_dir)
       not_git_dir = os.path.join(working_dir, "not_git")
       os.makedirs(not_git_dir)
       dir1 = os.path.join(working_dir, "dir1")
       os.makedirs(dir1)
       dir2 = os.path.join(working_dir, "dir2")
       os.makedirs(dir2)
       ... code to run the tests ...
```

You could wrap these `with` statements around the code to the unittest `run` method, but I think it would be better to use this inside of each individual test method, so that they do not reuse test repositories between different tests.

So, if you combined that with the suggested changes to the `setup_repos` method below, you would wind up with each test method looking something like:

```py
    def test_...(self)
        with tempfile.TemporaryDirectory() as working_dir:
            test_env = setup_repos(working_dir)
            ... code to perform the test...
```